google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionClientTests.java

```diff
@@ -503,4 +505,23 @@ public void onSessionCreateFailure(Throwable t, int createFailureForSessionCount
     }
     assertThat(returnedSessionCount.get()).isEqualTo(numSessions);
   }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testSessionNamePatterns() {
+    // Valid pattern for host session name
+    String host =
+        "projects/spanner-project/instances/spanner-instance/databases/test-db/sessions/abcd1234";
+    // Valid pattern for external host session name
+    String externalHost = "instances/default/databases/test-db/sessions/abcd1234";
+    try {
+      SessionId hostSession = SessionId.of(host);
+      assertEquals("abcd1234", hostSession.getName());
+      SessionId externalHostSession = SessionId.of(externalHost);
+      assertEquals("abcd1234", externalHostSession.getName());
+      // If no exceptions are thrown, the test will pass
+    } catch (IllegalArgumentException e) {
```

Comment:
nit: you can remove the entire try-catch block. If a (runtime) exception occurs during a test, then this test will be marked as failed with the error message. I.e. you will get the exact same behavior as you have now with the try-catch block, but with a bit less code and indentation.